### PR TITLE
Refactored Spinbox component & Spinbox Test cover 

### PR DESCRIPTION
**Pull request by JanMolcik**
#### Description, the reason for the PR
To fix Spinbox's unlimited max value and unknown error afterwards, I have implemented max value check and validation. On top of that, I have introduced test suit for current Spinbox functionality so that in the future it is easier to extend the component with more functionalities or change behaviour of existing.

#### Fixes issues
... 

#### License Agreement for contributions

- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
  - Improved quantity input in the Spinbox component with enforced minimum and maximum limits, integer-only input, immediate correction of invalid entries, and clearer UI feedback when limits are reached.
  - Added client-side validation to prevent integer overflow by enforcing a maximum cart quantity of 2,147,483,647.

- **Bug Fixes**
  - Input values in the Spinbox are now consistently validated and clamped within allowed bounds, preventing accidental over- or under-entry.
  - Restored last valid input value on invalid or empty entries to improve user experience.

- **Tests**
  - Added comprehensive automated tests for the Spinbox component, covering rendering, interactions, validation, edge cases, accessibility, and keyboard navigation.

- **Chores**
  - Enhanced test setup with new dependencies and configuration for improved testing capabilities.

----
:globe_with_meridians: Live Preview:
  - https://jm-product-count-error-ssp-3309.odin.shopsys.cloud
  - https://cz.jm-product-count-error-ssp-3309.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/vitest/components/Spinbox/Spinbox.test.tsx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        test('buttons have proper titles', () => {
+            const { container } = render(<Spinbox {...defaultProps} />);
+
+            const decreaseButton = container.querySelector('button[title="Decrease"]') as HTMLButtonElement;
+            const increaseButton = container.querySelector('button[title="Increase"]') as HTMLButtonElement;
+
+            expect(decreaseButton.getAttribute('title')).toBe('Decrease');
+            expect(increaseButton.getAttribute('title')).toBe('Increase');
+        });
+    });
```
Comment (chlebektomas) [inline project-base/storefront/vitest/components/Spinbox/Spinbox.test.tsx:344]: It would be nice to have a test for users to use the keyboard for navigation. Like Tab, Enter, Arrows and maybe if the user, for example, types some invalid string invalid.

**Review by chlebektomas — CHANGES_REQUESTED** [review]

File: project-base/storefront/vitest/components/Spinbox/Spinbox.test.tsx (hunk 1 above)
Comment (JanMolcik) [reply project-base/storefront/vitest/components/Spinbox/Spinbox.test.tsx:344]: Cool idea, will look into it.

**Review by chlebektomas — APPROVED** [review]
Nice, looks good!

**Review by pk16011990 — APPROVED** [review]
